File: iitj_assignments/assignment_1/c programs/1_student_details.c
#include<stdio.h>
#include<stdlib.h>

//student structure
struct student{
  char name[100];
  int roll_no;
  float marks;
};

//Insertion of student details
void insertion(struct student s[],int n)
{
   for(int i=0;i<n;i++)
   {
     //Name
     printf("enter %d th student name :",i+1);
     fflush(stdin);
     gets(s[i].name);

     //Roll Number
     printf("enter %d th student roll number :",i+1);
     scanf("%d",&s[i].roll_no);

     //Marks
     printf("enter %d th student marks :",i+1);
     scanf("%f",&s[i].marks);

   }
}

//Updation of student details
void update(struct student s[],int n)
{
  int stu_no,what;
  printf("Which student's detail you wan to update ? \n");
  scanf("%d",&stu_no);


  printf("what value you want to update ? \n 1. Press 1 for Name\n 2. Press 2 for Marks\n");
  scanf("%d",&what);
  //Invalid student number
  if(what>n || what<=0)
  {
    printf(" %d number of students present\n",n);
    return;
  }


  printf("Update Now..\n");
  //Name Updation
  if(what == 1)
  {
    fflush(stdin);
    gets(s[stu_no-1].name);
  }
    else
      //Marks Updation
      scanf("%f",&s[stu_no-1].marks);

}

//Displaying student details
void display(struct student s[],int n)
{
  printf("Name \tRoll Number\tMarks\n");
  for(int i=0;i<n;i++)
      printf("%s\t %d\t\t %f\n",s[i].name,s[i].roll_no,s[i].marks);
}
//Util function to show menu to the user
void print_menu()
{
   printf("1.Update details\n");
   printf("2.Display\n");
   printf("3.EXIT\n");
   printf("enter your choice :\n");

}

//main function
int main()
{
   int n,choice;
   //printf("*************S T A R T************\n");
   printf("How many students in the class ? \n");
   scanf("%d",&n);

   //creation
   struct student s[n];
   //insertion
   printf("Enter student details one by one...\n");
   insertion(s,n);

   while(1)
   {
     printf("**************************************************\n");
     print_menu();
     scanf("%d",&choice);
     switch(choice)
     {         //student details updation
        case 1:update(s,n);
               break;
                //display student details
        case 2:  display(s,n);
                 break;
                 //exit from program
        case 3:
                 exit(0);
        default:
                 printf("wrong choice\n");
                 break;
      }
    }
}

File: iitj_assignments/assignment_1/python programs/1_student_details.py
#!/usr/bin/env python
# coding: utf-8

# In[1]:


class student:
    
    def __init__(self,n,name,roll,marks):
        self.n =n
        self.name = name
        self.roll=roll
        self.marks = marks
    def display(self):
        print("Name : " ,self.name,"      Roll :",self.roll,"    Marks : ",self.marks)
    
    def update(self):
        print("1.Press 1 for name updation  \n 2.Press 2 for marks updation\n");
        what = int(input())
        if what ==1:
            print("Enter new name : ")
            self.name=input()
        else:
            print("Enter new marks : ")
            self.marks = int(input())
    


# In[3]:


students=[]
print("Enter number of students\n")
n=int(input())

#creation
for i in range(n):
    print("enter ",i+1," student details\n")
    print("Name :")
    name= input()
    print("Roll :")
    roll=int(input())
    print("Marks :")
    marks = float(input())
    
    students.append(student(n,name,roll,marks))

#display
for i in range(n):
    students[i].display()
    
#update
print("Which student details you want to update : \n")

stu_no =int(input())
if(stu_no<=0 or stu_no>n):
        print("Invalid student number\n");
else:
    students[i-1].update()

#display
for i in range(n):
    students[i].display()

